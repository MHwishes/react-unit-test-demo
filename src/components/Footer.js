import React from 'react';
import FilterLink from '../containers/FilterLink';
import {VisibilityFilters} from '../actions';

const Footer = () => (
    <footer className='footer'>
        <ul className="filters">
            <li>
                <FilterLink filter={VisibilityFilters.SHOW_ALL}>All</FilterLink>
            </li>
            <li>
                <FilterLink filter={VisibilityFilters.SHOW_COMPLETED}>Completed</FilterLink>
            </li>
            <li>
                <FilterLink filter={VisibilityFilters.SHOW_ACTIVE}>Active</FilterLink>
            </li>
        </ul>
    </footer>
);

export default Footer
