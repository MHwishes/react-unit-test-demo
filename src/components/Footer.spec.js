import React from "react";
import {shallow} from 'enzyme';
import Footer from './Footer'
import toJson from 'enzyme-to-json';

describe('<Footer/>', () => {
    it('render Footer component structure', () => {
        const wrapper = shallow(<Footer/>);
        expect(wrapper.find('li').length).toEqual(3);
        //Todo: please write <Footer/> component snapshot test
    });
});
