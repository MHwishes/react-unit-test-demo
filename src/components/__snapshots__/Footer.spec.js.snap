// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> render TodoList footer 1`] = `
<footer
  className="footer"
>
  <ul
    className="filters"
  >
    <li>
      <Connect(Link)
        filter="SHOW_ALL"
      >
        All
      </Connect(Link)>
    </li>
    <li>
      <Connect(Link)
        filter="SHOW_COMPLETED"
      >
        Completed
      </Connect(Link)>
    </li>
    <li>
      <Connect(Link)
        filter="SHOW_ACTIVE"
      >
        Active
      </Connect(Link)>
    </li>
  </ul>
</footer>
`;

exports[`shallow render todoList Footer component render TodoList footer 1`] = `
<footer
  className="footer"
>
  <ul
    className="filters"
  >
    <li>
      <Connect(Link)
        filter="SHOW_ALL"
      >
        All
      </Connect(Link)>
    </li>
    <li>
      <Connect(Link)
        filter="SHOW_COMPLETED"
      >
        Completed
      </Connect(Link)>
    </li>
    <li>
      <Connect(Link)
        filter="SHOW_ACTIVE"
      >
        Active
      </Connect(Link)>
    </li>
  </ul>
</footer>
`;
